nav2_map_server/src/map_3d/map_saver_3d.cpp

```diff
@@ -0,0 +1,192 @@
+/* Copyright (c) 2020 Shivam Pandey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "nav2_map_server/map_3d/map_saver_3d.hpp"
+
+#include <string>
+#include <memory>
+#include <future>
+#include <stdexcept>
+#include <functional>
+
+#include "sensor_msgs/msg/point_cloud2.hpp"
+
+using namespace std::placeholders;
+
+namespace nav2_map_server
+{
+
+MapSaver3D::MapSaver3D()
+: nav2_util::LifecycleNode("map_saver", "", true)
```

Comment:
Again, please sync-up with latest map saver source where the constructor has `("map_saver", "", false, options)` arguments. I want to draw your attention to the third "false" argument: RCLCPP internal node won't be created in this case. We do not need it moreover since new SingleThreadedExecutor with callback group model.